src/components/stake/StakeValidatorFilter.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+    <div class="validator-filter">
+        <div class="validator-filter-wrapper">
+            <button class="validator-switch" :class="state === FilterState.TRUST?'selected':''"
+                @click="ctrl.stateChange(FilterState.TRUST)">
+                {{ $t('TrustScore') }}
+            </button>
+            <button class="validator-switch" :class="state === FilterState.PAYOUT?'selected':''"
+                @click="ctrl.stateChange(FilterState.PAYOUT)">
+                {{ $t('Payout Time') }}
+            </button>
+            <button class="validator-switch" :class="state === FilterState.REWARD?'selected':''"
+                @click="ctrl.stateChange(FilterState.REWARD)">
+                {{ $t('Reward') }}
+            </button>
+            <span class="validator-search" @click="ctrl.stateChange(FilterState.SEARCH)">
+                <SearchIcon/>
+            </span>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref } from '@vue/composition-api';
+import SearchIcon from '../icons/Staking/SearchIcon.vue';
+
+enum FilterState {
+    TRUST,
+    PAYOUT,
+    REWARD,
+    SEARCH,
+}
+
+export default defineComponent({
+    setup() {
+        const state = ref(FilterState.TRUST);
+
+        return {
+            state,
+            FilterState,
+            ctrl: {
```

Comment:
There's no need to put methods into a separate object / namespace.
We usually don't do that.